src/MessageTrait.php

```diff
@@ -92,7 +92,7 @@ public function withHeader($header, $value): self
 
     public function withAddedHeader($header, $value): self
     {
-        if (!\is_string($header) || '' === $header) {
+        if (!\is_int($header) && (!\is_string($header) || '' === $header)) {
```

Comment:
[PHPStan is complaining](https://github.com/Nyholm/psr7/pull/149/checks?check_run_id=613771113) about this addition. And I think rightfully so. Removing this `is_int()` call does not break any tests. If we need this here, could you add a test case that requires it?

[PSR-7](https://www.php-fig.org/psr/psr-7/) only allows string values for `$header` here, unless I am [misreading something](https://github.com/php-fig/http-message/blob/efd67d1dc14a7ef4fc4e518e7dee91c271d524e4/src/MessageInterface.php#L145-L150):

```php
    /**
     * […snip…]
     * @param string $name Case-insensitive header field name to add.
     * @param string|string[] $value Header value(s).
     * @return static
     * @throws \InvalidArgumentException for invalid header names or values.
     */
    public function withAddedHeader($name, $value);
```

So we shouldn’t need to account for integers here. Any application calling `MessageInterface::withAddedHeader` (or `MessageInterface::withHeader` for that matter) MUST provide strings.

Reply:
I think this may need to be fixed [here](https://github.com/Nyholm/psr7-server/blob/e6a526e9170e6e33a13efc2b61703ca476b7ea68/src/ServerRequestCreator.php#L68) instead then. Since that's the function that passes integers to this method.

Reply:
Correct! As Nyholm/psr7-server can work with any PSR-7 implementation and not just this one it should follow the spec and make sure to only pass in string values there.

Reply:
Okay. See https://github.com/Nyholm/psr7-server/pull/36. That PR depends on this one to be merged.